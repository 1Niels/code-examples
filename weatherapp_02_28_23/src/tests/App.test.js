/**
* @jest-environment jsdom
*/

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
// import testing library functions
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import WeatherApp from '../App';
// import WeatherComponent from '../components/WeatherComponent';

// test that the UI matches the wireframes
test('renders Philadelphia link', () => {
  const { getByText } = render(<WeatherApp />);
  const linkElement = getByText(/Philadelphia/);
  expect(linkElement).toBeInTheDocument();
});

// snapshot testing

test('Weather app matches snapshot', () => {
  const component = renderer.create(<WeatherApp />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
/**
// test click event

test('The weather is shown after clicking on a link', async () => {
  // render the component
  render(<WeatherApp />);
  // fire a click on the a link with text Philadelphia'
  await userEvent.click(screen.getByText('Philadelphia'));
  // The weather box is shown and 'Temp' is visible
  const linkElement = screen.getByText(/Temp/);
  expect(linkElement).toBeInTheDocument();
});
*/