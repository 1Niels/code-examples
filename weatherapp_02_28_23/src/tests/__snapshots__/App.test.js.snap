// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather app matches snapshot 1`] = `
<div>
  <section>
    <h1>
      Cities
    </h1>
    <nav>
      <ul>
        <li>
          <a
            href="/weather/Philadelphia"
            onClick={[Function]}
          >
            Philadelphia
          </a>
        </li>
        <li>
          <a
            href="/weather/Boston"
            onClick={[Function]}
          >
            Boston
          </a>
        </li>
        <li>
          <a
            href="/weather/Mumbai"
            onClick={[Function]}
          >
            Mumbai
          </a>
        </li>
        <li>
          <a
            href="/weather/Dakar"
            onClick={[Function]}
          >
            Dakar
          </a>
        </li>
        <li>
          <a
            href="/weather/Shanghai"
            onClick={[Function]}
          >
            Shanghai
          </a>
        </li>
        <li>
          <a
            href="/weather/London"
            onClick={[Function]}
          >
            London
          </a>
        </li>
        <li>
          <a
            href="/weather/Montreal"
            onClick={[Function]}
          >
            Montreal
          </a>
        </li>
        <li>
          <a
            href="/weather/Paris"
            onClick={[Function]}
          >
            Paris
          </a>
        </li>
        <li>
          <a
            href="/custom/"
            onClick={[Function]}
          >
            Enter a city
          </a>
        </li>
      </ul>
    </nav>
    <article>
      <div
        className="App"
      >
        <label
          htmlFor="a"
        >
           
          Name:
          <input
            name="name"
            onChange={[Function]}
            type="text"
          />
        </label>
        <button
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
      </div>
    </article>
  </section>
</div>
`;
